src/KubernetesClient/Cache/ThreadSafeStore.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Threading;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace k8s.cache
+{
+    public class ThreadSafeStore : IThreadSafeStore
+    {
+        private static Mutex _storageLock;
+        private Dictionary<string, IKubernetesObject> _store;
+
+        public ThreadSafeStore() {
+            _store = new Dictionary<string, IKubernetesObject>();
+            _storageLock = new Mutex();
+        }
+        public void Add(string key, IKubernetesObject value)
+        {
+            _storageLock.WaitOne();           
+            try {                
+                if (_store.ContainsKey(key))
+                {
+                    _store[key] = value;
+                } else {
+                    _store.Add(key, value);            
+                }
+            } catch (Exception ex) {
+                Console.WriteLine(ex);    
```

Comment:
Don't swallow exceptions.

Reply:
Yep, moving out of WIP will involve that and tests at every level.